coriolis/tests/osmorphing/test_windows.py

```diff
@@ -0,0 +1,860 @@
+
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_get_worker_os_drive_path'
+    )
+    def test__add_dism_driver(self, mock_get_worker_os_drive_path):
+        mock_get_worker_os_drive_path.return_value = "C:"
+        result = self.morphing_tools._add_dism_driver(
+            mock.sentinel.driver_path)
+
+        self.assertEqual(result, self.conn.exec_command.return_value)
+        self.conn.exec_command.assert_called_once_with(
+            "C:\\Windows\\System32\\dism.exe",
+            ['/add-driver', '/image:%s' % self.os_root_dir,
+             '/driver:"%s"' % mock.sentinel.driver_path,
+             '/recurse', '/forceunsigned'])
+
+        mock_get_worker_os_drive_path.assert_called_once_with()
+
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_get_worker_os_drive_path'
+    )
+    def test__add_dism_driver_exception(self, mock_get_worker_os_drive_path):
+        mock_get_worker_os_drive_path.return_value = "C:"
+        self.conn.test_path.return_value = True
+        self.conn.exec_command.side_effect = CoriolisTestException
+
+        with self.assertLogs('coriolis.osmorphing.windows',
+                             level=logging.ERROR):
+            self.assertRaises(CoriolisTestException,
+                              self.morphing_tools._add_dism_driver,
+                              mock.sentinel.driver_path)
+        mock_get_worker_os_drive_path.assert_called()
+        self.conn.test_path.assert_called_once_with(
+            "C:\\Windows\\Logs\\DISM\\dism.log")
+
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_get_worker_os_drive_path'
+    )
+    def test__add_dism_driver_not_found(self, mock_get_worker_os_drive_path):
+        self.conn.test_path.return_value = False
+        self.conn.exec_command.side_effect = CoriolisTestException
+
+        with self.assertLogs('coriolis.osmorphing.windows',
+                             level=logging.WARN):
+            self.assertRaises(CoriolisTestException,
+                              self.morphing_tools._add_dism_driver,
+                              mock.sentinel.driver_path)
+        mock_get_worker_os_drive_path.assert_called()
+
+    def test__mount_disk_image(self):
+        result = self.morphing_tools._mount_disk_image(
+            mock.sentinel.path)
+
+        self.conn.exec_ps_command.assert_called_once_with(
+            "(Mount-DiskImage '%s' -PassThru | Get-Volume).DriveLetter" %
+            mock.sentinel.path)
+
+        self.assertEqual(result, self.conn.exec_ps_command.return_value)
+
+    def test__dismount_disk_image(self):
+        self.morphing_tools._dismount_disk_image(mock.sentinel.path)
+
+        self.conn.exec_ps_command.assert_called_once_with(
+            "Dismount-DiskImage '%s'" % mock.sentinel.path, ignore_stdout=True)
+
+    def test__expand_archive(self):
+        self.conn.test_path.return_value = True
+        destination = ':\\Windows\\destination'
```

Comment:
Even though this works, because this check passes:
```python
if destination.endswith(":\\") or ":\\Windows" in destination:
```
I would suggest using an actual valid Windows path. Something like this:
```python
destination = 'C:\\Windows\\destination'
```